server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/RepositoryServiceV1.java

```diff
@@ -0,0 +1,179 @@
+import com.linecorp.centraldogma.server.internal.storage.StorageExistsException;
+import com.linecorp.centraldogma.server.internal.storage.StorageNotFoundException;
+import com.linecorp.centraldogma.server.internal.storage.project.ProjectManager;
+
+/**
+ * Annotated service object for managing repositories.
+ */
+public class RepositoryServiceV1 extends AbstractService {
+
+    public RepositoryServiceV1(ProjectManager projectManager, CommandExecutor executor) {
+        super(projectManager, executor);
+    }
+
+    /**
+     * GET /projects/{projectName}/repos?status={status}
+     *
+     * <p>Returns the list of the repositories or removed repositories.
+     */
+    @Get("/projects/{projectName}/repos")
+    public CompletionStage<List<RepositoryDto>> listRepositories(@Param("projectName") String projectName,
+                                                                 @Param("status") Optional<String> status) {
+        checkProjectExists(projectName);
+        if (status.isPresent()) {
+            if (status.get().equalsIgnoreCase("removed")) {
+                return CompletableFuture.completedFuture(
+                        projectManager().get(projectName).repos().listRemoved().stream()
+                                        .map(name -> {
+                                            RepositoryDto r = new RepositoryDto();
+                                            r.setName(name);
+                                            return r;
+                                        }).collect(toList()));
+            } else {
+                // wrong status
+                throw new IllegalArgumentException("invalid status: " + status);
+            }
+        } else {
+            return CompletableFuture.completedFuture(
+                    projectManager().get(projectName).repos().list().values().stream()
+                                    .map(DtoConverter::convert)
+                                    .collect(toList()));
+        }
+    }
+
+    /**
+     * POST /projects/{projectName}/repos
+     *
+     * <p>Creates a new repository.
+     */
+    @Post("/projects/{projectName}/repos")
+    public CompletionStage<RepositoryDto> createRepository(@Param("projectName") String projectName,
+                                                           AggregatedHttpMessage message) {
+        checkProjectExists(projectName);
+
+        final JsonNode jsonNode;
+        try {
+            jsonNode = Jackson.readTree(message.content().toStringAscii()).get("name");
+        } catch (IOException e) {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+        checkArgument(jsonNode != null, "repository name should be non-null");
+        final String repoName = jsonNode.asText();
+        checkArgument(!isNullOrEmpty(repoName), "repository name should be non-null");
+
+        return execute(Command.createRepository(System.currentTimeMillis(), AuthenticationUtil.currentAuthor(),
```

Comment:
You can just use the factory method without timestamp parameter to let the upstream decide.